src/Microsoft.DotNet.Cli.Utils/CommandResolution/DefaultCommandResolver.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DefaultCommandResolver : CompositeCommandResolver
+    {
+        public static DefaultCommandResolver Create()
+        {
+            var environment = new EnvironmentProvider();
+            var packagedCommandSpecFactory = new PackagedCommandSpecFactory();
+
+            var platformCommandSpecFactory = default(IPlatformCommandSpecFactory);
```

Comment:
can't use var with null :smile: 

Is there a foreseeable downside to using default?

I like the suggestion to start with the generic though.